examples/vision/mixUp.py

```diff
@@ -0,0 +1,218 @@
+
+(Note that $lambda x_{i}$ and $lambda y_{i}$ are values with the [0, 1] range and are
+sampled from the [Beta distribution](https://en.wikipedia.org/wiki/Beta_distribution).)
+
+The technique is quite holistically named - we are literally mixing up the features and
+their corresponding labels. Implementation-wise it's simple. mixup can be extended to a
+variety of data modalities such as computer vision, natural language processing, speech,
+and so on.
+
+This example requires TensorFlow 2.4 or higher, as well as TensorFlow Probability, which
+can be installed using the following command:
+"""
+
+"""shell
+pip install tensorflow-probability
+"""
+
+"""
+## Setup
+"""
+
+import numpy as np
+import tensorflow as tf
+import matplotlib.pyplot as plt
+import tensorflow.keras.layers as L
+from tensorflow.keras.utils import to_categorical
+
+import tensorflow_probability as tfp
+
+tfd = tfp.distributions
+
+"""
+## Prepare the dataset
+
+In this example, we will be using the [FashionMNIST](https://research.zalando.com/welcome/mission/research-projects/fashion-mnist/) dataset. But this same recipe can
+be used for other classification datasets as well.
+"""
+
+(x_train, y_train), (x_test, y_test) = tf.keras.datasets.fashion_mnist.load_data()
+
+x_train = x_train.astype("float32") / 255.0
+x_train = np.reshape(x_train, (-1, 28, 28, 1))
+y_train = to_categorical(y_train, 10)
+
+x_test = x_test.astype("float32") / 255.0
+x_test = np.reshape(x_test, (-1, 28, 28, 1))
+y_test = to_categorical(y_test, 10)
+
+"""
+## Define hyperparameters
+"""
+
+AUTO = tf.data.AUTOTUNE
+BATCH_SIZE = 64
+EPOCHS = 5
+
+"""
+## Convert the data into TensorFlow `Dataset` objects
+"""
+
+train_ds_one = (
+    tf.data.Dataset.from_tensor_slices((x_train, y_train))
+    .shuffle(BATCH_SIZE * 100)
+    .batch(BATCH_SIZE)
+)
+train_ds_two = (
+    tf.data.Dataset.from_tensor_slices((x_train, y_train))
+    .shuffle(BATCH_SIZE * 100)
+    .batch(BATCH_SIZE)
+)
+# Because we will be mixing up the images and their corresponding labels, we will be
+# combining two shuffled datasets from the same training data.
+train_ds = tf.data.Dataset.zip((train_ds_one, train_ds_two))
+
+test_ds = tf.data.Dataset.from_tensor_slices((x_test, y_test)).batch(BATCH_SIZE)
+
+"""
+## Define the mixup technique function
+
+To perform the mixup routine, we create new virtual datasets using the training data from
+the same dataset, and apply a $\lambda$ value within the [0, 1] range sampled from a [Beta
+distribution](https://en.wikipedia.org/wiki/Beta_distribution) — such that, for example, `new_x = lambda * x1 + (1 - lambda) * x2` (where
+`x1` and `x2` are images) and the same equation is applied to the labels as well.
+"""
+
+
+def mix_up(ds_one, ds_two, alpha=0.2):
+    # Unpack two datasets
+    images_one, labels_one = ds_one
+    images_two, labels_two = ds_two
+    batch_size = tf.shape(images_one)[0]
+
+    # Sample lambda and reshape it to do the mixup
+    l = tfd.Beta(0.2, 0.2).sample(batch_size)
```

Comment:
Since Beta distribution is not directly available inside TensorFlow, I used it. Since Colab Notebooks come with TensorFlow Probability pre-installed I don't think it will be a problem but I do acknowledge your point.

Reply:
You could just copy how it's implemented in TFP. Unfortunately we imperatively have to stop depending on TFP (we will scrub it from other examples that currently use TFP).